src/marqo/tensor_search/tensor_search.py

```diff
@@ -853,29 +869,69 @@ def get_query_vectors_from_jobs(
                     ) for content, weight in ordered_queries
                 ]
                 # TODO how do we ensure order?
-                weighted_vectors = [np.asarray(vec) * weight for vec, weight, content in vectorised_ordered_queries]
+                collected_weights = [weight for _, weight, _ in vectorised_ordered_queries]
+                collected_vectors = [vec for vec, _, _ in vectorised_ordered_queries]
 
+            # Add context tensors
             context_tensors = q.get_context_tensor()
             if context_tensors is not None:
-                weighted_vectors += [np.asarray(v.vector) * v.weight for v in context_tensors]
+                collected_weights += [v.weight for v in context_tensors]
+                collected_vectors += [v.vector for v in context_tensors]
+
+            # Add context document vectors
+            context_documents = q.get_context_documents()
+            if context_documents:
+                context_doc_vectors = config.recommender.get_doc_vectors_from_ids(
+                    index_name=q.index.name,
+                    documents=context_documents.ids,
+                    tensor_fields=context_documents.parameters.tensorFields
+                )
 
-            for vector in weighted_vectors:
+                # Update weights and vectors list
+                for document_id, vector_list in context_doc_vectors.items():
+                    weight = context_documents.ids[document_id]
+                    # Per doc, add whole list of vectors, copy the doc weight for each
+                    collected_vectors.extend(vector_list)
+                    collected_weights.extend([weight] * len(vector_list))
+
+                # Save original doc ids for exclusion filtering
+                all_document_ids = list(context_documents.ids.keys())
+
+                # Determine default interpolation method using normalize embeddings if documents provided
+                if interpolation_method is None:
+                    interpolation_method = config.recommender.get_default_interpolation_method(q.index)
+            else:
```

Comment:
Is this duplicated with the line `interpolation_method = config.recommender.get_default_interpolation_method(q.index)`? Can we delete else and the line `interpolation_method = config.recommender.get_default_interpolation_method(q.index)` and always run the block to determine the interpolation here?

Reply:
Updated the get_default_interpolation_method method and removed redundant block. Just added the logic inside the method:
```
        if marqo_index.normalize_embeddings:
            if context_documents is not None:
                return InterpolationMethod.SLERP
            else:
                # NLERP is used to preserve existing search behavior with no context docs.
                return InterpolationMethod.NLERP
        else:
            return InterpolationMethod.LERP
```